py3status/modules/window_list.py

```diff
@@ -0,0 +1,228 @@
+    format_separator: show separator only if more than one (default ' ')
+    format_window: display format for windows
+        (default '\?max_length=30 [\?if=is_urgent&color=urgent \u2756'
+        |[\?if=is_focused&color=focused \u2756|\?color=unfocused \u2756]] {name}')
+
+Format placeholder:
+    {format_window}  format for windows
+    {floating}       number of floating windows      eg 3
+    {scratchpad}     number of scratchpad windows    eg 4
+    {urgent}         number of urgent windows        eg 2
+    {window}         number of windows               eg 11
+
+format_window placeholder:
+    {name}           window title name
+    is_floating      a boolean based on window data
+    is_focused       a boolean based on window data
+    is_scratchpad    a boolean based on window data
+    is_urgent        a boolean based on window data
+
+Requires:
+    i3ipc-python: an improved python library to control i3wm
+
+Examples:
+```
+# add colors
+window_list {
+    color_floating = '#3DCCB0'
+    color_focused = '#FFAE3D'
+    color_scratchpad = '#666666'
+    color_unfocused = '#3DAEE9'
+    color_urgent = '#FF48C4'
+}
+
+# show everything... excluding colors
+window_list {
+    button_next = 5
+    button_previous = 4
+
+    format = '{format_window} [\?color=#0ff&show FLOAT:] {floating} '
+    format += '[\?color=#0ff&show SCRATCHPAD:] {scratchpad} '
+    format += '[\?color=#0ff&show URGENT:] {urgent} '
+    format += '[\?color=#0ff&show WINDOW:] {window}'
+
+    format_window = '\?max_length=30 [\?if=is_urgent&color=urgent \u2756'
+    format_window += '|[\?if=is_focused&color=focused \u2756'
+    format_window += '|[\?if=is_scratchpad&color=scratchpad \u2756'
+    format_window += '|[\?if=is_floating&color=floating \u2756'
+    format_window += '|\?color=unfocused \u2756]]]] {name}'
+}
+```
+
+SAMPLE OUTPUT
+[
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' Netflix '},
+    {'color': u'#FF48C4', 'full_text': u'\u2756'},
+    {'full_text': u' #py3status - irssi'},
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' z420: ~ '},
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' ultrabug/py3status: py3stat '},
+    {'color': u'#FFAE3D', 'full_text': u'\u2756'},
+    {'full_text': u' window_list.py (~/src/py3st '},
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' mpd_status.py (~/src/py3stu '},
+    {'color': u'#3DAEE9', 'full_text': u'\u2756'},
+    {'full_text': u' (10) Facebook - Waterfox'},
+]
+
+"""
+
+from threading import Thread
+
+import i3ipc
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    button_focus = 1
+    button_next = None
+    button_previous = None
+    format = '{format_window}'
+    format_separator = ' '
+    format_window = u'\?max_length=30 [\?if=is_urgent&color=urgent \u2756' +\
+        '|[\?if=is_focused&color=focused \u2756|\?color=unfocused \u2756]] {name}'
```

Comment:
ok but I think you are resigning this module to a long struggle to get merged.

this does not sound like a feature currently needed, so could be added later as improvement.  I'm also of the view that there is a fundamental broken logic going on here, because I can be urgent and focused etc.